design/m1_push_to_file_design.md

```diff
@@ -0,0 +1,1163 @@
+| conjur.org/<br />conjur-configmap    | CONJUR_AUTHN_URL<br />CONJUR_URL<br />CONJUR_ACCOUNT<br />CONJUR_SSL_CERTIFICATE | ConfigMap containing Conjur connection information. |
+
+#### Annotations for Push-to-File Secrets Mappings
+
+##### Secrets Groups
+
+The configuration of the Secrets Provider push-to-file secrets mappings
+makes use of "secrets groups". A secrets group is a logical grouping of
+application secrets, typically belonging to a particular component of an
+application deployment (e.g. all secrets related to a backend database).
+
+As shown in the table below, all push-to-file annotations contain a reference
+to a secret group, using the form: `conjur.org/<parameter-name>.<group>`.
+
+There is a one-to-one correspondence between secrets groups and files that
+are written to the application Pod's Conjur secrets shared volume.
+
+##### Supported Annotations
+
+| Annotation                                     | Description |
+|------------------------------------------------|-------------|
+| conjur.org/conjur-secrets.{secret-group}       | List of secrets to be retrieved from Conjur. Each entry can be either:<ul><li>A Conjur variable path</li><li> A key/value pairs of the form `<alias>:<Conjur variable path>` where the `alias` represents the name of the secret to be written to the secrets file |
+| conjur.org/conjur-secrets-path.{secret-group}  | Prefix to use for all Conjur variable paths for a secrets group. Defaults to root policy path. |
+| conjur.org/secret-file-path.{secret-group}     | File path (including file name) for secrets file to be written. Default is `{secret-group}.{secret-file-type}`, e.g. `database.yaml`. |
+| conjur.org/secret-file-type.{secret-group}     | Allowed values:<ul><li>yaml (default)</li><li>json</li><li>dotenv</li><li>bash</li></ul>This setting is ignored when `conjur.org/secret-file-template.{secret-group}` is configured. |
+| conjur.org/secret-file-perms.{secret-group}    | File permissions for secrets file to be written, in octal format. Defaults to `660` |
+| conjur.org/secret-file-template.{secret-group} | Customized secret file template using the Golang templating language. |
+
+###### Example `conjur-secrets` List Without Aliases
+
+The following is an example of a `conjur.org/conjur-secrets.{secret-group}`
+Annotation that defines a flat (i.e. no aliases) list of Conjur variable paths
+for a hypothetical secrets group named `database`:
+
+```
+    conjur.org/conjur-secrets.database: |
+      - prod/backend/url
+      - prod/backend/port 
+      - prod/backend/password
+      - prod/backend/username
+```
+
+For `conjur-secrets` entries without aliases, the last word in the path is
+used as the application secret name. So for the example above, if
+`conjur.org/secret-file-type.database` is set to `yaml`, then the resulting
+secrets file that is written to the shared secrets volume might look similar
+to the following:
```

Comment:
```suggestion
no other annotations are set for the `database` secret group, then Secrets
Provider will write to a file `database.yaml` in the shared volume `/path/to/volume`
with contents that look like the following:
```

btw, what is the default mount path that the SP container will assume? I presume that we should have a default (it would correspond to default `secrets.mountPath` value in the helm chart section you have for SP volumes) and the `database.yaml` file would be written there (to the default place), unless the `conjur.org/conjur-secrets-path.` annotation is set

also, database.yaml or database.yml?